### PR TITLE
Add TestUtils.withRunGarn and use it in RunSpec

### PR DESCRIPTION
**Pull request by soenkehahn**

### CODE REVIEW

File: test/spec/RunSpec.hs [hunk 1]
```diff
@@ -9,265 +9,243 @@ import Data.String.Interpolate.Util (unindent)
...
+        [i|
+          import * as garn from "#{repoDir}/ts/mod.ts"
+
+          export const main = garn.shell("echo foobarbaz");
+        |]
+      output <- runGarn ["run", "main"]
+      stdout output `shouldBe` "foobarbaz\n"
+      exitCode output `shouldBe` ExitSuccess
+
+    it "propagates the exit status of the child process" $ \runGarn -> do
```
Comment (jkarni) [inline test/spec/RunSpec.hs:42]: I'm assuming the only change here is indentation?

File: test/spec/BuildSpec.hs [hunk 2]
```diff
@@ -8,32 +8,13 @@ import Development.Shake (CmdOption (EchoStdout), StdoutTrim (..), cmd, cmd_)
...
-          test runGarn'
-      )
-    $ spec
-
 spec :: Spec
 spec = do
   describe "build" $ do
     repoDir <- runIO getCurrentDirectory
-    wrap $ do
+    withRunGarn $ do
```
Comment (jkarni) [inline test/spec/BuildSpec.hs:17]: Maybe move this to immediately after `describe "build"` for consistency with the file below, or vice-versa? Definitely TIOLI, though

**Review by jkarni — APPROVED** [review]
The diff is a bit ugly, but I'm assuming the only thing changed is the obvious refactoring of `withRunGarn`, and using it? If so, LGTM

File: test/spec/BuildSpec.hs (hunk 2 above)
Comment (soenkehahn) [reply test/spec/BuildSpec.hs]: Yup, done.

File: test/spec/RunSpec.hs (hunk 1 above)
Comment (soenkehahn) [reply test/spec/RunSpec.hs:42]: Well, and removing the call to `onTestFailureLog` and calling the passed in `runGarn`. But yeah, it's a very mechanical refactoring.

**Comment by soenkehahn** [comment]
> The diff is a bit ugly, but I'm assuming the only thing changed is the obvious refactoring of `withRunGarn`, and using it? If so, LGTM

Yeah, exactly.

**Pull request merged**